src/App.jsx

```diff
@@ -18,14 +18,29 @@ function App() {
         "Lesson 1.7 Asynchronous Data, Conditional Rendering"
     ]
     const initialBlogs = initialTitles.map(title => titleToBlog(title))
-    const [blogTitles, setBlogTitles] = useState(getItem(LOCALSTORAGE_KEY) || initialBlogs)
-    const [searchText, setSearchText] = useState('');
-    const [focus, setFocus] = useState('search');
+    const [blogTitles, setBlogTitles] = useState([])
+    const [isLoading, setIsLoading] = useState(true)
+
+    useEffect(() => {
+        new Promise(
+            (resolve, reject) => {
+                setTimeout(() => {
+                    resolve(getItem(LOCALSTORAGE_KEY) || initialBlogs)
+                }, 2000)
+            }
+        ).then((blogTitlesFromLocalStorage) => {
```

Comment:
Javascript will run the callback function in the `.then` method when the promise is resolved. The callback function argument will be the argument passed into the `resolve` function in the promise.